code-name-taran/src/com/spaghettisoft/component/scrabble/TurnMakeWord.java

```diff
@@ -0,0 +1,185 @@
+			isWordOk = WordValidation.checkIfWordIsValidAndCouldBePlacedOnTheBoard(word, stave.getRow(), stave.getCol(), stave.getCh(), board, player);
+		}
+	}
+
+	private int drawWord(Player player, Board board, Bag bag) {
+		ArrayList<Stave> memoryStick = WordValidation.fillMemoryStick(word, stave.getRow(), stave.getCol(), stave.getCh(), board);
+		String playerLetters = Checks.takeLettersPlayerHasInHand(player);
+
+		writeTheWordOnTheBoard(memoryStick, playerLetters, board);
+//=============================================================================================================================================
+		String lettersThatRemainInPlayer = trimmLettersFromPlayerLetters(memoryStick, playerLetters, board);
+
+
+		if(lettersThatRemainInPlayer.equals("")) {
+			player.increasePoints(50);
+			System.out.println("Player used all 7 letters that has: +50 bonus points.");
+		}
+
+		String lettersTakyngSomeLettersFormTheBag = playerTakesNewLettersFromTheBag(lettersThatRemainInPlayer, bag);
+		ArrayList<Letter> letters = new ArrayList<>();
+		for (int i = 0; i < lettersTakyngSomeLettersFormTheBag.length(); i++) {
+			char ch = lettersTakyngSomeLettersFormTheBag.charAt(i);
+			int points = Constants.checkPointsForLetter(ch);
+			Letter letter = new Letter(ch, points, 0);
+			letters.add(letter);
+		}
+		player.setLetters(letters);
+		playerLetters = Checks.takeLettersPlayerHasInHand(player);
+//=============================================================================================================================================
+
+		int wordPoints = calculatePointsThePlayerMade(word, stave.getRow(), stave.getCol(), stave.getCh(), board);
+		System.out.println("Word - " + word + ", points - " + wordPoints);
+
+		return wordPoints;
+	}
+
+	private int calculatePointsThePlayerMade(String word, int row, int col, char orientation, Board board) {
+		int wordPoints = 0;
+		int multiplierWord = 1;
+		for (int i = 0; i < word.length(); i++) {
+			char ch = ' ';
+			int points = Constants.checkPointsForLetter(word.charAt(i));
+			int multiplierLetter = 0;
+			if(orientation == 'h') {
+				multiplierLetter = board.getTiles()[row-1][col-1+i].getMultiplierLetter();
+				ch = board.getTiles()[row-1][col-1+i].getContent().charAt(2);
+			} else {
+				multiplierLetter = board.getTiles()[row-1+i][col-1].getMultiplierLetter();
+				ch = board.getTiles()[row-1+i][col-1].getContent().charAt(2);
+			}
+			if(ch == '!') {
+				points = 0;
+			}
+			wordPoints += points*multiplierLetter;
+		}
+		for (int i = 0; i < word.length(); i++) {
+			int multiplier = 1;
+			if(orientation == 'h') {
+				multiplier = board.getTiles()[row-1][col-1+i].getMultiplierWord();
+			} else {
+				multiplier = board.getTiles()[row-1+i][col-1].getMultiplierWord();
+			}
+			multiplierWord *= multiplier;
+		}
+		wordPoints *= multiplierWord;
+		return wordPoints;
+	}
+
+	private String playerTakesNewLettersFromTheBag(String playerLetters, Bag bag) {

```

Comment:
This is inappropriate name for a method. Maybe something like "replaceLetters" is more fitted : )